table/block_based/block_based_table_iterator.cc

```diff
@@ -557,5 +700,119 @@ void BlockBasedTableIterator::FindReadAheadSizeUpperBound() {
 
   block_prefetcher_.SetUpperBoundOffset(start_offset +
                                         total_bytes_till_upper_bound);
+
+  if (getenv("Print")) {
+    printf("FindReadAheadSizeUpperBound:End\n");
+  }
+}
+
+void BlockBasedTableIterator::BlockCacheLookupForReadAheadSize(
+    size_t readahead_size, size_t& updated_readahead_size) {
+  if (getenv("Print")) {
+    printf("BlockCacheLookupForReadAheadSize\n");
+  }
+  ClearBlockHandles();
```

Comment:
Shouldn't block_handles be empty at this point? All buffered blocks should have been consumed by now, or already cleared if there was a reseek?

Reply:
Yes it should be. Let me add the assert instead to be safe.

Reply:
Done!